runtime/list_test.go

```diff
@@ -92,6 +92,29 @@ func TestListCount(t *testing.T) {
 	}
 }
 
+func TestListRemove(t *testing.T) {
+	fun := newBuiltinFunction("TestListRemove", func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		rem, raised := GetAttr(f, ListType.ToObject(), NewStr("remove"), nil)
+		if raised != nil {
+			return nil, raised
+		}
+		if _, raised := rem.Call(f, args, nil); raised != nil {
+			return nil, raised
+		}
+		return args[0], nil
+	}).ToObject()
+	cases := []invokeTestCase{
+		{args: wrapArgs(newTestList(1, 2, 3), 2), want: newTestList(1, 3).ToObject()},
+		{args: wrapArgs(NewList()), wantExc: mustCreateException(TypeErrorType, "'remove' of 'list' requires 2 arguments")},
+		{args: wrapArgs(NewList(), 1), wantExc: mustCreateException(ValueErrorType, "list.remove(x): x not in list")},
```

Comment:
That's true but we also target 1.6 and 1.7 :)

Additionally, in my experience, even a 4x improvement would not be sufficient to justify its use in low level code. The underlying issue is that it does an allocation in most cases. Much of Grumpy's design and many optimizations revolve around avoiding allocations because they're so expensive.